Projects/DbCLI/CreateDb.py

```diff
@@ -321,144 +325,144 @@ def CreateDb(options, dataFilename='', supplier=None):
+        fpCurs.executemany('insert into %(s)s values (?,?,?)' % args.pharm2DTableName, pharm2DRows)
         pharm2DRows = []
         fpConn.Commit()
       if len(gobbi2DRows):
-        fpCurs.executemany('insert into %s values (?,?,?)' % options.gobbi2DTableName, gobbi2DRows)
+        fpCurs.executemany('insert into %(s)s values (?,?,?)' % args.gobbi2DTableName, gobbi2DRows)
         gobbi2DRows = []
         fpConn.Commit()
       if len(morganRows):
-        fpCurs.executemany('insert into %s values (?,?,?)' % options.morganFpTableName, morganRows)
+        fpCurs.executemany('insert into %(s)s values (?,?,?)' % args.morganFpTableName, morganRows)
         morganRows = []
         fpConn.Commit()
 
-    if not options.silent and not i % 500:
-      logger.info('  Done: %d' % (i))
+    if not args.silent and not i % 500:
+      logger.info('  Done: %(d)s' % (i))
 
   if len(pairRows):
-    pairCurs.executemany('insert into %s values (?,?,?,?)' % options.pairTableName, pairRows)
+    pairCurs.executemany('insert into %(s)s values (?,?,?,?)' % args.pairTableName, pairRows)
     pairRows = []
     pairConn.Commit()
   if len(fpRows):
-    fpCurs.executemany('insert into %s values (?,?,?)' % options.fpTableName, fpRows)
+    fpCurs.executemany('insert into %(s)s values (?,?,?)' % args.fpTableName, fpRows)
     fpRows = []
     fpConn.Commit()
   if len(layeredRows):
-    fpCurs.executemany('insert into %s values (?,?,%s)' % (options.layeredTableName, layeredQs),
+    fpCurs.executemany('insert into %(s)s values (?,?,%(s)s)' % (args.layeredTableName, layeredQs),
                        layeredRows)
     layeredRows = []
     fpConn.Commit()
   if len(descrRows):
-    descrCurs.executemany('insert into %s values (%s)' % (options.descrTableName, descrQuery),
+    descrCurs.executemany('insert into %(s)s values (%(s)s)' % (args.descrTableName, descrQuery),
                           descrRows)
     descrRows = []
     descrConn.Commit()
   if len(pharm2DRows):
-    fpCurs.executemany('insert into %s values (?,?,?)' % options.pharm2DTableName, pharm2DRows)
+    fpCurs.executemany('insert into %(s)s values (?,?,?)' % args.pharm2DTableName, pharm2DRows)
     pharm2DRows = []
     fpConn.Commit()
   if len(gobbi2DRows):
-    fpCurs.executemany('insert into %s values (?,?,?)' % options.gobbi2DTableName, gobbi2DRows)
+    fpCurs.executemany('insert into %(s)s values (?,?,?)' % args.gobbi2DTableName, gobbi2DRows)
     gobbi2DRows = []
     fpConn.Commit()
   if len(morganRows):
-    fpCurs.executemany('insert into %s values (?,?,?)' % options.morganFpTableName, morganRows)
+    fpCurs.executemany('insert into %(s)s values (?,?,?)' % args.morganFpTableName, morganRows)
     morganRows = []
     fpConn.Commit()
 
-  if not options.silent:
+  if not args.silent:
     logger.info('Finished.')
 
 
 if __name__ == '__main__':
-  options, args = parser.parse_args()
-  if options.loadMols:
+  args = parser.parse_args()
+  if args.loadMols:
```

Comment:
While I can understand why you changed the name of the variable, it doesn't address the actual issue and distracts from the real changes in the code. This makes it very hard to review.

Reply:
@gedeck I have reverted the format strings, realized that I went wrong with %s and %d, while only %default needed to be changed to %(default)s.

This is a part of the documentation of argparse, regarding the upgrade from optparse -
Replace (options, args) = parser.parse_args() with args = parser.parse_args() and add additional ArgumentParser.add_argument() calls for the positional arguments. Keep in mind that what was previously called options, now in the argparse context is called args.

May you please clarify on *additional ArgumentParser.add_argument() calls for the positional arguments*?

Reply:
You can use `options = parser.parse_args()`

A command line applications has two types of arguments similar to a function in Python. The positional arguments are the ones that are not associated with a specific option, e.g. `--delimiter`. The positional arguments are what usually comes at the end of the command line, e.g. in `python file.py` file.py is the positional argument. 

`optparse` was differentiating them in the result of `parse_args` while `argparse` adds them to the others using a name. The `CreateDB.py` file has only one positional argument to specify the `<filename>`. In the original code, you can find the line:
```
dataFilename = args[0]
```
This is where it is accessed. In `argparse`, you need to add an argument without dashes:
```
parser.add_argument('filename')
```
`argparse` assigns the first positional argument to it and you will be able to use
```
args = parser.parse_args()
dataFilename = args.filename
```
Does this clarify your question?

Reply:
Yeah, thanks!

Changing them now.